packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -195,6 +195,10 @@ export class Cluster extends Resource implements ICluster {
       autoScalingGroup.addUserData('echo ECS_AWSVPC_BLOCK_IMDS=true >> /etc/ecs/ecs.config');
     }
 
+    if (options.spotInstanceDraining) {
```

Comment:
should we verify this ASG has spot capacity at all or maybe it doesn't really matter.

Reply:
The classic ASG with launch configuration could be either `all on-demand`  or `all spot` and [ASG with mixed purchase options](https://aws.amazon.com/blogs/aws/new-ec2-auto-scaling-groups-with-multiple-instance-types-purchase-options/) is another story. 

Is it possible to determine if this ASG is `all spot` in the `cluster.ts`? If yes, then I prefer we enable this feature by default for the `all spot` ASG unless user explicitly turns it off. And for `all on-demand` ASG we turn it off.

At this moment, I leave this open to let user specify explicitly.

Reply:
Oh, I think we can check if user specify the `spotPrice` property. If yes, then we turn on this by default unless user explicitly turn it off. 

Thoughts?

Reply:
If the default is `off` then that might be a breaking change which should be pushed off. I think eladb made a good point that we should check spot capacity first but the default from my perspective should remain the same.

Reply:
fixed in https://github.com/aws/aws-cdk/pull/4360/commits/ae46f3c3f25c66258a1d92baab29c93044140e23